frontend/src/components/Button/Button.ts

```diff
@@ -1,9 +1,93 @@
 import Component from '../../core/Component';
 import './Button.scss';
 
-class Button extends Component {
+import { add, close, chevronDown } from '../../../assets/icons';
+
+export enum ButtonRole {
+  Medium = 'medium',
+  Large = 'large',
+  Fab = 'fab',
+  Tab = 'tap',
+  Category = 'category',
+  Status = 'status',
+  Add = 'add',
+  withClose = 'with-close',
+}
+
+interface ButtonProps {
+  text?: string;
+  className?: string;
+  type?: 'submit' | 'reset' | 'button';
+  role?: ButtonRole;
+  width?: string;
+  onClickHandler?: (event: Event) => void;
+  onCloseHandler?: (event: Event) => void;
+}
+
+interface ButtonState {}
+
+const buttonClassMap = {
+  [ButtonRole.Medium]: 'with-text with-font-small with-size-medium',
+  [ButtonRole.Large]: 'with-text with-font-medium with-size-large',
+  [ButtonRole.Fab]: 'with-round-full with-icon',
+  [ButtonRole.Tab]: 'with-text with-font-small',
+  [ButtonRole.Category]: 'with-round with-text with-font-small',
+  [ButtonRole.Status]: 'with-icon with-text with-font-small',
+  [ButtonRole.Add]: 'with-icon with-font-small with-size-medium',
+  [ButtonRole.withClose]:
+    'with-icon with-text with-close with-font-small with-size-medium',
+};
+
+class Button extends Component<ButtonProps, ButtonState> {
```

Comment:
이 부분 Component 클래스를 수정했어서 ButtonState가 빈 객체일 경우 따로 선언하지 않아줘도 될 것 같네요 !
```suggestion
class Button extends Component<ButtonProps> {
```

Reply:
넵, 확인해서 반영했습니다!